### DISCUSSION TITLE
Any way to run pyodide totally quiet ?

### DISCUSSION THREAD

**Discussion by svax974** [pyodide/pyodide#4039] [General]
Hello, is there a way to prevent any output ?
I tried to pass empty handlers for stdout and err both in loadPyodide and with pyodide.setStdout/err but I still have all the pyodide messages like the 'Loading' ones for the modules.

Thanks !

**Comment by hoodmane** [comment]
If you pass `{ messageCallback : ()=>{}, errorCallback: ()=>{} }` as options to [`loadPackage`](https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.loadPackage) then it will be quiet.

**Comment by svax974** [comment]
Thanks, that helps indeed !
Any way to have micropip to be quiet too in Javascript ?

**Reply by hoodmane** [reply]
I don't think there is a "public" way (@ryanking13?) but you can make it silent by monkeypatching `loadPackage` before importing micropip:
```js
const pyodide = await loadPyodide();
origLoadPackage = pyodide.loadPackage;
pyodide.loadPackage = function (pkgs, options) {
    return origLoadPackage(pkgs, {messageCallback: () => {}, errorCallback: () => {}, ...options});
}
await pyodide.loadPackage("micropip");
const micropip = await pyodide.pyimport("micropip")
await micropip.install("attrs");
```

**Reply by ryanking13** [reply]
> I don't think there is a "public" way (@ryanking13?)

We don't have a public way, but I think we can consider adding some options to control it.

**Comment by svax974** [comment]
Thanks, I'll try that asap.
